packages/compass-indexes/src/components/name-column/name-column.jsx

```diff
@@ -1,38 +1,38 @@
 import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { spacing, css, Body, IndexKeys } from '@mongodb-js/compass-components';
 
-import IndexDefinitionType from '../index-definition-type';
+const containerStyles = css({
+  paddingLeft: spacing[4],
+  paddingBottom: spacing[3],
+  position: 'relative',
+  width: '35%',
+});
 
-import classnames from 'classnames';
-import styles from './name-column.module.less';
+const indexKeyStyles = css({
+  marginTop: spacing[1],
+});
 
-/**
- * Component for the name column.
- */
 class NameColumn extends PureComponent {
   static displayName = 'NameColumn';
 
   static propTypes = {
     index: PropTypes.object.isRequired,
   };
 
-  /**
-   * Render the name column.
-   *
-   * @returns {React.Component} The name column.
-   */
   render() {
+    const indexKeys = {};
+    const indexName = this.props.index.name;
+    this.props.index.fields.serialize().forEach(({ field, value }) => {
+      indexKeys[field] = value;
+    });
```

Comment:
Why are we converting existing array to object to use it in the component as array again?

Reply:
due to aggregation explain where we have an object, i tried to align this with that implementation. however, after your comment, i made `keys` an array of object and aligned aggregation explain with that.